pkg/internal/controllers/certificaterequests.go

```diff
@@ -122,19 +121,19 @@ func addCertificateRequestController(ctx context.Context, opts Options) error {
 		// Watch CertificateRequestPolicies. If a policy is created or updated,
 		// then we need to process all CertificateRequests that do not yet have an
 		// approved or denied condition since they may be relevant for the policy.
-		Watches(&source.Kind{Type: new(policyapi.CertificateRequestPolicy)}, handler.EnqueueRequestsFromMapFunc(enqueueRequestFromMapFunc)).
+		Watches(&policyapi.CertificateRequestPolicy{}, handler.EnqueueRequestsFromMapFunc(enqueueRequestFromMapFunc)).
```

Comment:
`Watches` now expects `client.Object` instead of `source.Source`.
This change matches how c/r calls `Watches` in tests https://github.com/kubernetes-sigs/controller-runtime/pull/2135/commits/dc62ed5dadc6ea2d3e22546368b939a2db1edfb2#diff-64658a075291eb3beb906fd3850a3477413c70ed6bc8fb02590708ded0d6f3baR499